src/de/mod.rs

```diff
@@ -723,6 +723,10 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
 
         self.parser.skip_ws()?;
 
+        if self.parser.exts.contains(Extensions::EXPLICIT_STRUCT_NAMES) && name.len() == 0 {
```

Comment:
With this, you are checking if the serde-given struct name is empty or not, but not whether the user has provided it or not. In line 721 just above, when we call `self.parser.consume_struct_name(name)?;`, the struct name is consumed (if it is there we eat it, if it's not we do nothing). I think that the check should rather be in this function such that continuing when the struct name isn't given is no longer allowed with the extension.

Reply:
I realized this while reading the codebase before I saw this, but thanks. I've implemented this logic in `Parser::consume_struct_name` and created a test for it.